File: AtlasWeb/Core/Modules/Tibbi_Surec/Manipulasyon_Modulu/AudiologyFormViewModel.ts
//$A2D6766F
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { AudiologyObject } from "NebulaClient/Model/AtlasClientModel";
import { AudiologyDiagnosis } from "NebulaClient/Model/AtlasClientModel";
import { AudiologyDiagnosisDef } from "NebulaClient/Model/AtlasClientModel";
import { Guid } from 'NebulaClient/Mscorlib/Guid';
import { Type } from 'NebulaClient/ClassTransformer';

export class AudiologyFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.AudiologyFormViewModel, Core';
    @Type(() => AudiologyObject)
    public _AudiologyObject: AudiologyObject = new AudiologyObject();
    @Type(() => AudiologyDiagnosis)
    public AudiologyDiagnosisGridList: Array<AudiologyDiagnosis> = new Array<AudiologyDiagnosis>();
    @Type(() => AudiologyDiagnosisDef)
    public AudiologyDiagnosisDefs: Array<AudiologyDiagnosisDef> = new Array<AudiologyDiagnosisDef>();
    
    @Type(() => Object)
    public _selectedDiagnosis: Object;
    @Type(() => AudiologyDiagnosisObject)
    public DiagnosisArray: Array<AudiologyDiagnosisObject> = new Array<AudiologyDiagnosisObject>();

}

export class AudiologyDiagnosisObject {
    @Type(() => Guid)
    public DiagnosisObjectID: Guid;
    @Type(() => Guid)
    public DiagnosisDefObjectID: Guid;
    public Name: string;
 
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Manipulasyon_Modulu/AudiologyFormViewModel.cs
//$6F9B2294
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using System.Collections.Generic;

namespace Core.Controllers
{
    public partial class AudiologyObjectServiceController
    {
        partial void PreScript_AudiologyForm(AudiologyFormViewModel viewModel, AudiologyObject audiologyObject, TTObjectContext objectContext)
        {
            viewModel.DiagnosisArray = new AudiologyDiagnosisObject[0];

            List<AudiologyDiagnosisObject> diagnosisList = new List<AudiologyDiagnosisObject>();

            foreach(AudiologyDiagnosis diagnosis in audiologyObject.AudiologyDiagnosis)
            {
                AudiologyDiagnosisObject d = new AudiologyDiagnosisObject();
                d.DiagnosisDefObjectID = diagnosis.AudiologyDiagnosisDef.ObjectID;
                d.DiagnosisObjectID = diagnosis.ObjectID;
                d.Name = diagnosis.AudiologyDiagnosisDef.DiagnosisName;
                diagnosisList.Add(d);
            }

            viewModel.DiagnosisArray = diagnosisList.ToArray();
        }

    }
}

namespace Core.Models
{
    public partial class AudiologyFormViewModel: BaseViewModel, IManipulationFormBaseObjectViewModel
    {

        public AudiologyDiagnosisObject[] DiagnosisArray;
       

        public void AddManipulationFormBaseObjectViewModelToContext(TTObjectContext objectContext)
        {
            foreach (AudiologyDiagnosisObject diagnosis in this.DiagnosisArray)
            {
                if (diagnosis.DiagnosisObjectID == Guid.Empty)
                {
                    var audiologyDiagnosis = new AudiologyDiagnosis(objectContext);
                    var audiologyDiagnosisDef = (AudiologyDiagnosisDef)objectContext.GetObject(diagnosis.DiagnosisDefObjectID, "AudiologyDiagnosisDef");
                    audiologyDiagnosis.AudiologyDiagnosisDef = audiologyDiagnosisDef;
                    audiologyDiagnosis.AudiologyObject = this._AudiologyObject;

                }
            }

            objectContext.AddToRawObjectList(this._AudiologyObject);
            objectContext.ProcessRawObjects();
        }

        public void SetManipulationReport(Manipulation manipulation)
        {
            manipulation.Report = this._AudiologyObject.AudiologyReport;
        }
    }

    public class AudiologyDiagnosisObject
    {
        public Guid DiagnosisObjectID;
        public Guid DiagnosisDefObjectID;
        public string Name;
    }
}
